util/base_parser.h

```diff
@@ -1,10 +1,13 @@
 // SPDX-FileCopyrightText: 2021 Robin Lindén <dev@robinlinden.eu>
+// SPDX-FileCopyrightText: 2022 David Zero <zero-one@zer0-one.net>
 //
 // SPDX-License-Identifier: BSD-2-Clause
 
 #ifndef UTIL_BASE_PARSER_H_
 #define UTIL_BASE_PARSER_H_
 
+#include "util/string.h"
+
 #include <algorithm>
```

Comment:
This doesn't need `<algorithm>` or `<array>` now that you've removed `is_space`.